### ISSUE TITLE
python3Packages.typer does not support click v8

### ISSUE THREAD

**Issue by alexvorobiev** [NixOS/nixpkgs#129479]
**Describe the bug**
typer is incompatible with click 8: https://github.com/tiangolo/typer/issues/280

**Steps to reproduce**
```
$ nix-shell -p python3Packages.typer
...
ERROR: Could not find a version that satisfies the requirement click<7.2.0,>=7.1.1 (from typer)
ERROR: No matching distribution found for click<7.2.0,>=7.1.1
builder for '/nix/store/lzz3rbni4xfpksmqlk8filkkch5z2gaw-python3.8-typer-0.3.2.drv' failed with exit code 1
```

**Notify maintainers**
@winpat 

**Metadata**
```yaml
 - system: `"x86_64-linux"`
 - host os: `Linux 3.10.0-1160.25.1.el7.x86_64, Red Hat Enterprise Linux Server, 7.9 (Maipo)`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.3.12`
 - channels(root): `"nixpkgs-21.11pre300649.3e0ce8c5d47"`
 - nixpkgs: `/root/.nix-defexpr/channels/nixpkgs`
```
Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: pythonPackages.typer
# a list of nixos modules affected by the problem
module:
```

**Comment by graham33** [comment]
I have a version of typer in my NUR that uses some WIP changes to support click v8 (see https://github.com/tiangolo/typer/pull/302) and disables the tests to sidestep remaining issues: https://github.com/graham33/nur-packages/blob/master/pkgs/typer/default.nix

With that I was able to get a build that I could use with nixos master for some downstream dependencies I needed (authcaptureproxy for homeassistant).  HTH

**Comment by tiangolo** [comment]
I just released Typer `0.4.0`, which is compatible with **both** Click 7 and Click 8. :tada:
